packages/beacon-state-transition/src/allForks/util/epochContext.ts

```diff
@@ -129,14 +129,14 @@ export function createEpochContext(
   let exitQueueEpoch = computeActivationExitEpoch(currentEpoch);
   let exitQueueChurn = 0;
 
-  const effectiveBalancesArr: number[] = [];
+  const validators = readonlyValuesListOfLeafNodeStruct(state.validators);
+  const validatorCount = validators.length;
+
+  const effectiveBalanceIncrements = new Uint8Array(getEffectiveBalanceIncrementsByteLen(validatorCount));
```

Comment:
in what case `effectiveBalanceIncrements` length is different from `validatorCount`? could it be `new Uint8Array(validatorCount)`

Reply:
effectiveBalanceIncrements will be equally or greater than validatorCount. The `getEffectiveBalanceIncrementsByteLen()` modulo is used to reduce the frequency at which its Uint8Array is recreated. if effectiveBalanceIncrements has length greater than validatorCount it's not a problem since those values would never be accessed.

Reply:
that's great, please drop a comment in that function